### PR TITLE
Ensure trigger payload is within specified size limit

### PR DESCRIPTION
**Pull request by bjustin-ibm**
Batch messages until any of the following conditions are met:
- there are no more messages in the topic
- 2 seconds have passed since the last time firing
- _**new**_: the currently consumed message would exceed the allowed payload size. When this happens, the current message is squirreled away for inclusion in the next poll cycle.

### CODE REVIEW

**Comment by bjustin-ibm** [comment]
@dubeejw please review

File: provider/consumer.py [hunk 1]
```diff
@@ -278,8 +287,20 @@ def __pollForMessages(self):
                 if (message is not None):
                     if not message.error():
                         logging.debug("Consumed message: {}".format(str(message)))
-                        messageSize += len(message.value())
-                        messages.append(message)
+                        messageSize = self.__sizeMessage(message)
+                        if totalPayloadSize + messageSize > payload_limit:
+                            if len(messages) == 0:
+                                logging.error('[{}] Single message at offset {} exceeds payload size limit. Skipping this message!'.format(self.trigger, message.offset()))
```
Comment (dubeejw) [inline provider/consumer.py]: Can the user alerted of the message being dropped?

**Comment by dubeejw** [comment]
@bjustin-ibm, possible to test batching. Also would be nice to doc the batching behavior or at least the max message limit.

**Comment by bjustin-ibm** [comment]
> possible to test batching

@dubeejw Batch limit is tested here: https://github.com/apache/incubator-openwhisk-package-kafka/pull/174/files#diff-739f35fdbcde166788378e5daa12611bR194

File: provider/consumer.py (hunk 1 above)
Comment (bjustin-ibm) [inline provider/consumer.py]: We don't currently have a nice mechanism for this. We are toying with the idea of exposing the trigger health info to the user or, even logs related to their triggers. That will be handled separately along with info related to automatic trigger disablement/deletion.

**Comment by bjustin-ibm** [comment]
> Also would be nice to doc the batching behavior

Agreed. I'll update the batching section of the README.

**Comment by bjustin-ibm** [comment]
@dubeejw README updated.

**Comment by dubeejw** [comment]
@bjustin-ibm, need test that verifies that messages that exceed the payload limit are dropped.

**Comment by bjustin-ibm** [comment]
> need test that verifies that messages that exceed the payload limit are dropped

@dubeejw Done: 23016b7cd52bdd99d8ec283a8ffeb45c73c44218

**Comment by bjustin-ibm** [comment]
@csantanapr please merge.

**Comment by bjustin-ibm** [comment]
@csantanapr Actually... give me a minute to manually squash these commits.

**Comment by bjustin-ibm** [comment]
@csantanapr Ok done. Please merge when Travis finishes.

**Pull request merged**